src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -517,6 +517,62 @@ private static void ArrayTypeCheck_Helper(object obj, void* elementType)
             }
         }
 
+        // Helpers for boxing
+        [DebuggerHidden]
+        private static object? Box_Nullable(IntPtr handle, ref byte nullableData)
+        {
+            MethodTable* srcMT = new TypeHandle((void*)handle).AsMethodTable();
+            return BoxNullableImpl(srcMT, ref nullableData);
+        }
+
+        [DebuggerHidden]
+        internal static object? BoxNullableImpl(MethodTable* srcMT, ref byte nullableData)
```

Comment:
This split into `Box_Nullable` and `BoxNullableImpl` should not be needed. It is fine for the JIT helper to take `MethodTable*` argument.